### DISCUSSION TITLE
Shadow in sampler ar viewer

### DISCUSSION THREAD

**Discussion by khyati18** [sceneview/sceneview#279] [Q&A]
Hi, I am not getting shadow on any model in samples ar-model-viewer. How can I get it?

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, AR shadows are supported through ARCore's environmental HDR lighting mode:

```kotlin
ARSceneView(
    sessionConfiguration = { _, config ->
        config.lightEstimationMode = Config.LightEstimationMode.ENVIRONMENTAL_HDR
    }
) {
    AnchorNode(anchor = anchor) {
        ModelNode(modelInstance = model)
    }
}
```

`ENVIRONMENTAL_HDR` provides realistic lighting and shadows based on the real environment. The shadow falls on the AR plane automatically. For best results, ensure the model has proper shadow-casting materials and the scene has good real-world lighting.
